packages/react/src/components/Toolbar/ToolbarItem.tsx

```diff
@@ -101,6 +105,20 @@ class ToolbarItem extends UIComponent<WithAsProp<ToolbarItemProps>, ToolbarItemS
         {childrenExist(children) ? children : Icon.create(icon)}
       </ElementType>
     )
+
+    if (popup) {
```

Comment:
Somehow I feel that the `popup` should not be treated as a regular shorthand prop, just because it is tightly in correlation with the element it is used in. At least we can introduce correct typings for allowing users to specify only the things that make sense, and allow it only to be defined as object/string. I really think it's confusing now that we will allow the user to specify some jsx element there - but it won't be used as `content` of the `Popup`, but rather will be treated as `Popup` and it will end up being modified if the `trigger` or `children` props are used. Let me know what are your thoughts.

Reply:
We can use the `Omit` generic to omit the children and trigger props from the `Popup`